easycv/pipeline.py

```diff
@@ -26,78 +31,206 @@ class Pipeline:
 
     def __init__(self, source, name=None):
         if isinstance(source, list):
-            self.forwards = Pipeline._calculate_forwards(source)
+            self.forwards = {}
 
             self.arguments = source[0].arguments if source else {}
             self.outputs = source[-1].outputs if source else {}
 
             self._name = name if name else "pipeline"
             self._transforms = deepcopy(source)
+            self.required = {}
+            self.optional = {}
+            self._arguments = {}
+            self.initialize()
 
         elif isinstance(source, str) and os.path.isfile(source):
-            try:
-                with open(source, "rb") as f:
-                    saved = pickle.load(f)
-                    if isinstance(saved, Pipeline):
-                        self._name = name if name else saved.name
-                        self._transforms = saved.transforms()
-                    else:
-                        raise InvalidPipelineInputSource()
-            except pickle.UnpicklingError:
-                raise InvalidPipelineInputSource() from None
+            with open(source, "r") as file:
+                source = yaml.load(file, Loader=yaml.FullLoader)
```

Comment:
Codacy found an issue: [Use of unsafe yaml load. Allows instantiation of arbitrary objects. Consider yaml.safe_load().](https://app.codacy.com/gh/easycv/easycv/pullRequest?prid=6255509)